packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js

```diff
@@ -17,47 +18,104 @@ import noMatchesFor from '@salesforce/label/c.quantic_NoMatchesFor';
 import collapseFacet from '@salesforce/label/c.quantic_CollapseFacet';
 import expandFacet from '@salesforce/label/c.quantic_ExpandFacet';
 
+/** @typedef {import("coveo").FacetState} FacetState */
+/** @typedef {import("coveo").Facet} Facet */
+/** @typedef {import("coveo").FacetValue} FacetValue */
+/** @typedef {import("coveo").SearchEngine} SearchEngine */
+
 const displayOptions = {
   checkbox: 'checkbox',
   link: 'link'
 }
 
+/**
+ * A facet is a list of values for a certain field occurring in the results, ordered using a configurable criteria (e.g., number of occurrences).
+ * A `QuanticFacet` displays a facet of the results for the current query.
+ * @example
+ * <c-quantic-facet engine-id={engineId} facet-id="myFacet" field="filetype" label="File Type" number-of-values="5" sort-criteria="occurrences" no-search display-values-as="link" is-collapsed></c-quantic-facet>
+ */
 export default class QuanticFacet extends LightningElement {
-  /** @type {import("coveo").FacetState} */
-  // @ts-ignore
-  @track state = {
-    values: [],
-  };
-
-  /** @type {string} */
+  /**
+   * The ID of the engine instance with which to register.
+   * @api
+   * @type {string}
+   */
+  @api engineId;
+  /** 
+   * An unique ID used to identify the facet instance.
+   * Defaults to the facet label.
+   * @api
+   * @type {string}
+   * @defaultValue `(field)`
+   */
   @api facetId;
-  /** @type {string} */
+  /**
+   * The field whose values you want to display in the facet.
+   * @api
+   * @type {string}
+   */
   @api field;
-  /** @type {string} */
+  /**
+   * The non-localized label for the facet.
+   * @api
+   * @type {string}
+   */
   @api label = 'no-label';
-  /** @type {string} */
-  @api engineId;
-  /** @type {number} */
+  /**
+   * The number of values to request for this facet.
+   * Also determines the number of additional values to request each time this facet is expanded, and the number of values to display when this facet is collapsed.
+   * @api
+   * @type {number}
+   * @defaultValue `8`
+   */
   @api numberOfValues = 8;
-  /** @type  {import("coveo").FacetSortCriterion}*/
+  /**
+   * The sort criterion to apply to the returned facet values
+   * Possible values are:
+   *   - `score`
+   *   - `numeric`
+   *   - `occurences`
+   *   - `automatic`
+   * @api
+   * @type  {'score' | 'alphanumeric' | 'occurrences' | 'automatic'}
+   * @defaultValue `'automatic'`
+   */
   @api sortCriteria = 'automatic';
-  /** @type {boolean} */
+  /**
+   * Whether this facet should not contain a search box.
+   * @api
+   * @type {boolean}
+   * @defaultValue `false`
+   */
   @api noSearch = false;
-  /** @type {string} */
+  /**
+   * Whether to display the facet values as checkboxes (multiple selection) or links (single selection). Possible values are 'checkbox', 'link'.
+   * @api
+   * @type {string}
```

Comment:
Why not `@type {'checkbox' | 'link'}` instead?

Reply:
that's better